src/backend/egl/fence.rs

```diff
@@ -0,0 +1,156 @@
+//! TODO: Module docs
+use std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, OwnedFd};
+use std::sync::Arc;
+use std::time::Duration;
+
+use crate::backend::egl::{ffi, wrap_egl_call, EGLDisplay, Error};
+
+#[derive(Debug)]
+struct InnerEGLFence {
+    display: EGLDisplay,
```

Comment:
EGLDisplay stores quite a bit of stuff, like DrmFormats and extensions. I would expect this type to be rather small. Can we instead store an `Arc<EGLDisplayHandle>` (maybe even a `Weak`) and a `supports_from_native` bool or something?
Also please don't evaluate extensions for every fence creation, but create this bool once when initializing the `EGLDisplay`. This can't change for the lifetime of an `EGLDisplay` anyway.